woocommerce/woocommerce-android
# Integrate application passwords changes from FluxC

⚠️ Please don't merge before updating the fluxc hash to a `trunk` commit.
This PR depends on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2589

### Description
This PR does two tasks:

1. Integrates the application password changes from FluxC.
2. Initiates a simple API call to `/wc/v3/settings` on the app launch or site switching to check if the site supports application supports, and if it doesn't, it will report a Tracks event.

This will be useful to investigate on how many sites have the feature disabled on their sites, and decide on the better approach to handle it.

### Testing instructions
Launch the app or change the site, then confirm the following:

1. For a site that supports Application Passwords, the tracks should log:

```
 🔵 Tracked: application_passwords_test_initiated
 🔵 Tracked: application_passwords_available
```

2. For a site that has Application Passwords disabled (for example using this [plugin](https://wordpress.org/plugins/disable-application-passwords/))), the tracks should log:

```
🔵 Tracked: application_passwords_test_initiated
🔵 Tracked: application_passwords_not_available, Properties: {"network_status_code":501,"error_code":"application_passwords_disabled","error_message":"Application passwords are not available."}
```
